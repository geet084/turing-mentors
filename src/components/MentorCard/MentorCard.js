import React, { Component } from 'react';
import { Link } from 'react-router-dom';

export class MentorCard extends Component {
  render() {
    const { id, first_name, last_name, location, tech_skills } = this.props;
    
    return (
      <Link to={`/mentors/${id}`}>
        <div className="mentors-card">
          <p>{first_name} {last_name}</p>
          <div className="loc-lang">
            <span className="loc">
              {location}
            </span>
            {
              tech_skills &&
              <span className="lang">
                {tech_skills[0]}
              </span>
            }
          </div>
        </div>
      </Link>
    )
  }
}

export default MentorCard;