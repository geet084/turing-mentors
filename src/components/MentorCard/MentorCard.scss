.mentors-card {
  box-shadow: 8px 9px 12px -11px rgb(72, 70, 72);
  margin: 15px;
  width: 275px;
  display: flex;
  flex-direction: column;
  background-color: lightgrey;
  align-items: center;
  border-radius: 15px;
  p, span {
    margin: 10px;
  }
}

.loc-lang {
  width: 100%;
  display: flex;
  justify-content: space-between;
  .loc {
    width: 65%;
    color: green;
  }
  .lang {
    width: 35%;
    color: red;
  }
}
