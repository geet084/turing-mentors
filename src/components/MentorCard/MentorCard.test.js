import React from 'react';
import { MentorCard } from './MentorCard';
import { shallow } from 'enzyme';

describe('MentorCard', () => {
  let wrapper;

  it('should match the snapshot if no tech skills are passed in', () => {
    wrapper = shallow(<MentorCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if tech skills are passed in', () => {
    wrapper = shallow(<MentorCard tech_skills={'javascript'} />);
    expect(wrapper).toMatchSnapshot();
  });
});