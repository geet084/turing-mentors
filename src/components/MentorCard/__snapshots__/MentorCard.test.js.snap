// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorCard should match the snapshot if no tech skills are passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MentorCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="mentors-card"
      >
        <p>
           
        </p>
        <div
          className="loc-lang"
        >
          <span
            className="loc"
          />
        </div>
      </div>,
      "to": "/mentors/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
             
          </p>,
          <div
            className="loc-lang"
          >
            <span
              className="loc"
            />
          </div>,
        ],
        "className": "mentors-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="loc"
              />,
              undefined,
            ],
            "className": "loc-lang",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "loc",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="mentors-card"
        >
          <p>
             
          </p>
          <div
            className="loc-lang"
          >
            <span
              className="loc"
            />
          </div>
        </div>,
        "to": "/mentors/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
               
            </p>,
            <div
              className="loc-lang"
            >
              <span
                className="loc"
              />
            </div>,
          ],
          "className": "mentors-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="loc"
                />,
                undefined,
              ],
              "className": "loc-lang",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "loc",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`MentorCard should match the snapshot if tech skills are passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MentorCard
    tech_skills="javascript"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="mentors-card"
      >
        <p>
           
        </p>
        <div
          className="loc-lang"
        >
          <span
            className="loc"
          />
          <span
            className="lang"
          >
            j
          </span>
        </div>
      </div>,
      "to": "/mentors/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
             
          </p>,
          <div
            className="loc-lang"
          >
            <span
              className="loc"
            />
            <span
              className="lang"
            >
              j
            </span>
          </div>,
        ],
        "className": "mentors-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="loc"
              />,
              <span
                className="lang"
              >
                j
              </span>,
            ],
            "className": "loc-lang",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "loc",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "j",
                "className": "lang",
              },
              "ref": null,
              "rendered": "j",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="mentors-card"
        >
          <p>
             
          </p>
          <div
            className="loc-lang"
          >
            <span
              className="loc"
            />
            <span
              className="lang"
            >
              j
            </span>
          </div>
        </div>,
        "to": "/mentors/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
               
            </p>,
            <div
              className="loc-lang"
            >
              <span
                className="loc"
              />
              <span
                className="lang"
              >
                j
              </span>
            </div>,
          ],
          "className": "mentors-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="loc"
                />,
                <span
                  className="lang"
                >
                  j
                </span>,
              ],
              "className": "loc-lang",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "loc",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "j",
                  "className": "lang",
                },
                "ref": null,
                "rendered": "j",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
